toolkit/cli/node-commands/README.md

```diff
@@ -0,0 +1,219 @@
+  --stake-pool-pub-key 0x702b81ab2e86cf73a87062af1eb0da666d451976d9d91c63a119ed94e6a33dc0 \
+  --mc-epoch-number 586
+```
+
+**Important Notes:**
+- If registration was included in Cardano block in epoch N, it should be visible when querying epoch N+1 or later
+- If registration doesn't appear after a few minutes, debugging may be required
+
+#### 3. Ariadne Parameters (`ariadne-parameters`)
+Retrieves Ariadne protocol parameters that are effective at a specific mainchain epoch. These parameters control various aspects of the Partner Chain consensus mechanism.
+
+**Usage:**
+```bash
+partner-chains-node ariadne-parameters --mc-epoch-number 586
+```
+
+**Note:** Parameters become effective two epochs after the block containing their change.
+
+#### 4. Registration Signatures (`registration-signatures`)
+Generates cryptographic signatures required for Partner Chain committee candidate registration.
+
+**Usage:**
+```bash
+partner-chains-node registration-signatures [OPTIONS]
+```
+
+#### 5. Address Association Signing (`sign-address-association`)
+Creates signatures for associating addresses between different chains or contexts.
+
+**Usage:**
+```bash
+partner-chains-node sign-address-association [OPTIONS]
+```
+
+#### 6. Block Producer Metadata Signing (`sign-block-producer-metadata`)
+Signs block producer metadata for submission to the runtime, enabling participation in block production.
+
+**Usage:**
+```bash
+partner-chains-node sign-block-producer-metadata [OPTIONS]
+```
+
+#### 7. Smart Contracts (`smart-contracts`)
+Provides a comprehensive suite of commands for interacting with Partner Chain smart contracts deployed on Cardano.
+
+**Usage:**
+```bash
+partner-chains-node smart-contracts [SUBCOMMAND]
+```
+
+#### 8. Setup Wizards (`wizards`)
+Interactive text-based wizards that guide users through the setup and configuration of Partner Chains.
+
+**Usage:**
+```bash
+partner-chains-node wizards [SUBCOMMAND]
+```
+
+## Architecture
+
+### Command Structure
+
+The crate is built around the `PartnerChainsSubcommand` enum, which defines all available commands. Each command is implemented as a separate struct with its own configuration and execution logic.
+
+```rust
+pub enum PartnerChainsSubcommand<RuntimeBindings, PartnerchainAddress> {
+    SidechainParams(SidechainParamsCmd),
+    RegistrationStatus(RegistrationStatusCmd),
+    AriadneParameters(AriadneParametersCmd),
+    // ... other commands
+}
+```
+
+### Generic Design
+
+The crate is designed with generics to support different runtime configurations and address types:
+
+- `RuntimeBindings`: Specifies the Partner Chain runtime bindings
+- `PartnerchainAddress`: Defines the address format used by the specific Partner Chain
+
+### CLI Integration
+
+All commands implement the `CliConfiguration` trait from Substrate's CLI framework, ensuring consistent behavior and configuration management across the Partner Chains ecosystem.
+
+## Dependencies
+
+### Core Dependencies
```

Comment:
Nobody would update this when he needs new dependency or removes one. I don't see a reason to repeat Cargo.toml.